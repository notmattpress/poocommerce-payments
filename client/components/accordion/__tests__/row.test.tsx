/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { AccordionRow } from '../';

describe( 'AccordionRow', () => {
	test( 'renders with default props', () => {
		const { container } = render(
			<AccordionRow>Test Content</AccordionRow>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders with custom className', () => {
		const { container } = render(
			<AccordionRow className="custom-class">Test Content</AccordionRow>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders with complex content', () => {
		const { container } = render(
			<AccordionRow>
				<div>
					<h3>Title</h3>
					<p>Description</p>
				</div>
			</AccordionRow>
		);
		expect( container ).toMatchSnapshot();
	} );
} );
