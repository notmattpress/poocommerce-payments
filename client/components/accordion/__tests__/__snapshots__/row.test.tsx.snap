// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionRow renders with complex content 1`] = `
<div>
  <div
    class="wcpay-accordion__row"
  >
    <div>
      <h3>
        Title
      </h3>
      <p>
        Description
      </p>
    </div>
  </div>
</div>
`;

exports[`AccordionRow renders with custom className 1`] = `
<div>
  <div
    class="wcpay-accordion__row custom-class"
  >
    Test Content
  </div>
</div>
`;

exports[`AccordionRow renders with default props 1`] = `
<div>
  <div
    class="wcpay-accordion__row"
  >
    Test Content
  </div>
</div>
`;
